### PR TITLE
network: disable weak ciphers explicitly

### PR DESCRIPTION
**Pull request by Chilledheart**

### CODE REVIEW

File: src/network-mgr.cpp [hunk 1]
```diff
@@ -2,12 +2,99 @@
...
+        else if (name == "MD5")
+            return true;
+        else if (name == "DES")
+            return true;
+        else if (name == "DSS")
+            return true;
+        else if (name == "EXP")
+            return true;
+        else if (name == "NULL")
+            return true;
```
Comment (lins05) [inline src/network-mgr.cpp]: It'll be more readable to construct a `QSet<String>` and use `QSet.contains()` than this kind of lots of if..else statements.

**Comment by lins05** [comment]
@Chilledheart Any reference document on which ciphers suite to use?

**Comment by Chilledheart** [comment]
@lins05 Yes, I think we do have some. 

For example, in the [RFC7465](https://tools.ietf.org/html/rfc7465), RC4 cipher is found insecure and not recommended to use. For a complete list of choice of these ciphers, you can refer to https://raymii.org/s/tutorials/Strong_SSL_Security_On_nginx.html .

File: src/network-mgr.cpp (hunk 1 above)
Comment (Chilledheart) [inline src/network-mgr.cpp]: @lins05 sorry, but I can't agree with you about using `QSet` and `QHashMap` everywhere. It just makes no sense to use `QSet` here but wastes a lot of CPU time if we can use plain `vector` instead. (Internally, the methods of `QSet` and `QHashMap` call hundred times of instructions than plain `vector` do, cost much more memory and don't scale well)

**Comment by shoeper** [comment]
See also https://www.rfc-editor.org/rfc/rfc7525.txt

**Comment by Chilledheart** [comment]
@shoeper Thanks for providing this information!

**Pull request merged**